### PR TITLE
Rename and rework build tests

### LINKED ISSUE #1034: Fix existing `dcpy` test file `test/lifecycle/builds/build.py`
While working on #1021, I noticed that the test file `dcpy/test/lifecycle/builds/build.py` doesn't appear to be included when running CICD. Most likely, it's due to the file name not containing the word "test". 

Goal: Rename the file and fix any failing tests.

### PR DESCRIPTION
**Pull request by fvankrieken**
Damon has inspired me with our newfound test coverage.

Closes #1034 

We should definitely expand on this - `builds` is very much not tested. But figured it'd be nice to get our existing tests running

### CODE REVIEW

**Comment by fvankrieken** [comment]
@sf-dcp - codecov edits the existing comment as you push commits

**Review by sf-dcp — APPROVED** [review]
Nice! Ty for fixing this

File: dcpy/lifecycle/builds/upload.py
Comment (fvankrieken) [inline dcpy/lifecycle/builds/upload.py:1]: This empty file was checked in - figured we should just remove

**Comment by fvankrieken** [comment]
Oops sorry @sf-dcp , I kept working on this not thinking that I had already requested review. Just added one more test - I'll get this merged after tests pass and leave at current coverage for now

**Review by damonmcc — APPROVED** [review]

**Pull request merged**